src/Shaders.cpp

```diff
@@ -1,13 +1,25 @@
 #include "Shaders.h"
 
 #include <fstream>
-#include <string>
+#include <string.h>
 #include <vector>
 
+#define EXTENSION_LENGTH 3
 
-GLuint LoadShaders(const char * vertex_file_path,const char * fragment_file_path)
+
+GLuint LoadShadersByName(const char * name)

```

Comment:
Honestly, this feels wrong and I might delete this function and just have RenderEngine.Init() require 2 args. We'll see how this develops as you choose libs and we work through ogre.